src/app/pages/home/containers/home-page.component.ts

```diff
@@ -124,20 +124,25 @@ export class HomePageComponent implements OnDestroy {
   }
 
   displayTaskCalendar() {
-    this.firebaseAnalytics.logEvent('click', { button: 'show_task_calendar' })
+    this.logEvent('click', { button: 'show_task_calendar' })
     this.showCalendar = !this.showCalendar
   }
 
   openSettingsPage() {
     this.navCtrl.push(SettingsPageComponent)
-    this.firebaseAnalytics.logEvent('click', { button: 'open_settings' })
+    this.logEvent('click', { button: 'open_settings' })
   }
 
   openClinicalTasksPage() {
-    this.firebaseAnalytics.logEvent('click', { button: 'open_clinical_tasks' })
     this.navCtrl.push(ClinicalTasksPageComponent)
+    this.logEvent('click', { button: 'open_clinical_tasks' })
   }
 
+  logEvent(event: string, params: any) {
```

Comment:
Should this be done for other pages/services too ?

Reply:
yes. I guess basically everywhere. Otherwise it breaks the workflow on browser. I tried to check it in the FirebaseAnalyticsService, but does not help. Better to wrap it in a method in the page/component.

There will be some other changes related firebase analytics. so we can wait

Reply:
Ok thanks